src/fidesops/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -0,0 +1,276 @@
+    CONSENT_REQUEST_VERIFY,
+    V1_URL_PREFIX,
+)
+from fidesops.ops.common_exceptions import (
+    FunctionalityNotConfigured,
+    IdentityVerificationException,
+)
+from fidesops.ops.core.config import config
+from fidesops.ops.models.email import EmailConfig
+from fidesops.ops.models.privacy_request import (
+    Consent,
+    ConsentRequest,
+    ProvidedIdentity,
+    ProvidedIdentityType,
+)
+from fidesops.ops.schemas.email.email import (
+    EmailActionType,
+    FidesopsEmail,
+    SubjectIdentityVerificationBodyParams,
+)
+from fidesops.ops.schemas.privacy_request import Consent as ConsentSchema
+from fidesops.ops.schemas.privacy_request import (
+    ConsentPreferences,
+    ConsentPreferencesWithVerificationCode,
+    ConsentRequestResponse,
+    VerificationCode,
+)
+from fidesops.ops.schemas.redis_cache import Identity
+from fidesops.ops.service.email.email_dispatch_service import dispatch_email_task
+from fidesops.ops.service.privacy_request.request_runner_service import (
+    generate_id_verification_code,
+)
+from fidesops.ops.tasks import EMAIL_QUEUE_NAME
+from fidesops.ops.util.api_router import APIRouter
+
+router = APIRouter(tags=["Consent"], prefix=V1_URL_PREFIX)
+
+logger = logging.getLogger(__name__)
+
+
+@router.post(
+    CONSENT_REQUEST,
+    status_code=HTTP_200_OK,
+    response_model=ConsentRequestResponse,
+)
+def create_consent_request(
+    *,
+    db: Session = Depends(get_db),
+    data: Identity,
+) -> ConsentRequestResponse:
+    """Creates a verification code for the user to verify access to manange consent preferences."""
+    if not config.redis.enabled:
+        raise FunctionalityNotConfigured(
+            "Application redis cache required, but it is currently disabled! Please update your application configuration to enable integration with a redis cache."
+        )
+
+    if not config.execution.subject_identity_verification_required:
+        raise FunctionalityNotConfigured(
+            "Subject identity verification is required, but it is currently disabled! Please update your application configuration to enable subject identity verification."
+        )
+
+    if not data.email:
+        raise HTTPException(HTTP_400_BAD_REQUEST, detail="An email address is required")
+
+    identity = ProvidedIdentity.filter(
+        db=db,
+        conditions=(
+            (ProvidedIdentity.field_name == ProvidedIdentityType.email)
+            & (ProvidedIdentity.hashed_value == ProvidedIdentity.hash_value(data.email))
+        ),
+    ).first()
+
+    if not identity:
+        provided_identity_data = {
+            "privacy_request_id": None,
+            "field_name": "email",
+            "encrypted_value": {"value": "test@email.com"},
+        }
+        identity = ProvidedIdentity.create(db, data=provided_identity_data)
+
+    consent_request_data = {
+        "provided_identity_id": identity.id,
+    }
+    consent_request = ConsentRequest.create(db, data=consent_request_data)
```

Comment:
I don't fully understand the need for this extra `ConsentRequest` table, and why `ProvidedIdentity` and `Consent` are not sufficient. Is it because we want to cache these verification codes under a separate `ConsentRequest` id, instead of under a `ProvidedIdentity` id since the ProvidedIdentity has multiple purposes?

Reply:
I had the same question. The reason was we might need to track different consent request data like dates and times.